### PR TITLE
M11

### PR DESCRIPTION
**Pull request by Dani2525**

### CODE REVIEW

File: .env.template
Comment (JackMeadDev) [inline .env.template:7]: We want to remove the values from this, but we don't want to just lose everything in the file and have it be empty! The idea is that this file gives a new dev a quick reminder of what variables need setting, specifies the format, and gives some sensible default values where possible
```suggestion
# Flask server configuration.
FLASK_APP=todo_app/app
FLASK_ENV=development

# Change the following values for local development.
SECRET_KEY=replace-with-a-secret-key
PRIMARY_CONNECTION=replace-with-your-connection-string
```

File: README.md
Comment (JackMeadDev) [inline README.md:41]: We've got your secrets here again! We don't want these ever in our codebase, and we should rotate the CosmosDB password and the client_secret now that they've been exposed

File: todo_app/app.py
Comment (JackMeadDev) [inline todo_app/app.py:38]: Nice, it would be good to have similar on your `complete_item` function too (unless we're intentionally allowing readers to do that)

File: todo_app/templates/index.html
Comment (JackMeadDev) [inline todo_app/templates/index.html:1]: This might tell writers they can write, but it's not much good for telling non-writers that they can't which is quite confusing!

Instead, what we probably want to do is only show the appropriate options to each person, e.g. for your "new item" form then we only want to show it to writers:
```
 {% if view_model.user_role == 'writer' %}
  <form action= "/createnewcard" method="post" > 
    <input type="text" name = "title">
    <input type="submit" value= "Add your new item in the box">
  </form>
  {% endif %}
```

**Review by JackMeadDev — CHANGES_REQUESTED** [review]
Nice work, I was able to follow your instructions and I can see that your tests are passing & your authentication flow with GitHub is working, but there are a couple of things we should complete before we merge:
* You've got all the setup in place, but you're not currently hiding any options from read-only users; we should be hiding e.g. the form for creating new items from users who don't have permissions (see below)
* Your client secret and database password are publicly available in your Readme, so we should delete those and also rotate (i.e. revoke and create new) both of those!

As a bonus but not critical here - when I visit your webapp in Azure it looks like you've deployed this image but not updated the environment variables/config there so it's not redirecting correctly; that might just be something we get working as part of M12 but it would be nice to have both apps working while they both exist

File: .env.template
Comment (Dani2525) [reply .env.template:7]: done

File: README.md
Comment (Dani2525) [reply README.md:41]: yes deleted now :)

File: .env.template
Comment (Dani2525) [reply .env.template:7]: done

File: todo_app/app.py
Comment (Dani2525) [reply todo_app/app.py:38]: yes that's more reasonable, done

File: todo_app/templates/index.html
Comment (Dani2525) [reply todo_app/templates/index.html:1]: done, also added to /complete_item

File: .env.template
Comment (Dani2525) [reply .env.template:7]: done

File: README.md
Comment (CorndelDevOps) [inline README.md:40]: Secrets are still here - you will want to make sure to document where to get these from rather than putting them into the version control.

Make sure any secrets are removed from the PR

**Review by CorndelDevOps — COMMENTED** [review]
Thank you for the updates.

I can see that you removed some previously committed secrets but there are still secrets in your `README.md` you will need to get rid of. Do this before completing the PR.

I've tried accessing either of your App Service deployment URLs.
* https://daniphilipwebapp.azurewebsites.net - I get 404
* https://dani-m12-todoapp.azurewebsites.net - I get redirected to localhost.
You will need to fix your app configuration. You can hold off to module 12 to do this.

**Pull request merged**
